table/table_test.cc

```diff
@@ -4490,8 +4490,7 @@ TEST_P(BlockBasedTableTest, DISABLED_TableWithGlobalSeqno) {
         user_props[ExternalSstFilePropertyNames::kVersion].c_str());
     global_seqno = DecodeFixed64(
         user_props[ExternalSstFilePropertyNames::kGlobalSeqno].c_str());
-    global_seqno_offset =
-        props->properties_offsets[ExternalSstFilePropertyNames::kGlobalSeqno];
+    global_seqno_offset = props->external_sst_file_global_seqno_offset;
```

Comment:
Question:
I tried to understand why this test is disabled ...

>... RocksDB deduces global_seqno from the MANIFEST while reading from an SST. 

Does this refer to `GetGlobalSequenceNumber ` logics in [here](https://github.com/facebook/rocksdb/blob/main/table/block_based/block_based_table_reader.cc#L490-L523) ?

> RocksDB does not store global_seqno in the SST file any more

The wording here is a bit confusing since it seems to me that external sst file global seqno still stores in SST table in the field `ExternalSstFilePropertyNames::kGlobalSeqno`...

And debbuger throws an segment fault in `[return table_reader->NewIterator(..](https://github.com/facebook/rocksdb/pull/9212/files#diff-3bed9ef8e04fd53a5be84539e52cc6565a7883a25ea46ba77ca01c2a62b85007R4519)`  when we called `GetTableInternalIter();` the second time [here](https://github.com/facebook/rocksdb/pull/9212/files#diff-3bed9ef8e04fd53a5be84539e52cc6565a7883a25ea46ba77ca01c2a62b85007R4528). Interesting thing about it is that my debugger can't even step in the  `NewIterator()` to pin down exact point of the exception ..

Reply:
SstFileWriter creates files with a property named kGlobalSeqno. At that point the global seqno is unknown so its value is some default value (I think zero).

In the old days, IngestExternalFile() would always overwrite that property's value with the global seqno determined at the time of ingestion. However it was problematic for WS where small I/Os are costly, plus our I/O pattern can be truly append-only if we got rid of the overwrite of kGlobalSeqno property's value.

So now we only overwrite the value if `write_global_seqno` is true. Otherwise that property is present but ignored, in which case the global seqno is determined from MANIFEST instead. Yes, that is the code in `GetGlobalSequenceNumber()` -- `largest_seqno` comes from MANIFEST and is used as the global seqno when the property value in the final was left at its default (zero).